### PR TITLE
PLT-7177: Change the default email frequency to 15 minutes if batching is enabled on the server.

### PR DESCRIPTION
**Pull request by MusikPolice**
#### Summary
Changed default email notification interval to 15 minutes when batching has been enabled in the system console. In the future, we should consider improving synchronization between preference defaults set in the web app and on the server, so that it's not possible for them to get out of sync. 

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-7177

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

File: app/email_batching.go [hunk 1]
```diff
@@ -140,13 +140,14 @@ func (job *EmailBatchingJob) checkPendingNotifications(now time.Time, handler fu
 		// get how long we need to wait to send notifications to the user
```
Comment (MusikPolice) [inline app/email_batching.go]: The code in this file is only executed if batching has been enabled in the System Console, so unlike on the UI, we don't have to check this preference. We can just assume that it is active and use the appropriate default value

File: model/preference.go [hunk 2]
```diff
@@ -33,7 +33,9 @@ const (
 
 	PREFERENCE_CATEGORY_NOTIFICATIONS = "notifications"
 	PREFERENCE_NAME_EMAIL_INTERVAL    = "email_interval"
-	PREFERENCE_DEFAULT_EMAIL_INTERVAL = "30" // default to match the interval of the "immediate" setting (ie 30 seconds)
+
+	PREFERENCE_EMAIL_INTERVAL_IMMEDIATE       = "30"  // the "immediate" setting is actually 30s
+	PREFERENCE_EMAIL_INTERVAL_FIFTEEN_MINUTES = "900" // fifteen minutes is 900 seconds
```
Comment (MusikPolice) [inline model/preference.go]: Ideally, these default values would be shared between server and client, but I have already burned a lot of time on this ticket. Maybe an improvement for the future?

File: webapp/components/admin_console/email_settings.jsx [hunk 3]
```diff
@@ -136,11 +134,11 @@ export default class EmailSettings extends AdminSettings {
                         <FormattedHTMLMessage
                             key='admin.email.enableEmailBatchingDesc'
                             id='admin.email.enableEmailBatchingDesc'
-                            defaultMessage='When true, users can have email notifications for multiple direct messages and mentions combined into a single email, configurable in <b>Account Settings > Notifications</b>.'
+                            defaultMessage='When true, users will have email notifications for multiple direct messages and mentions combined into a single email. Batching will occur at a default interval of 15 minutes, configurable in Account Settings > Notifications.'
```
Comment (MusikPolice) [inline webapp/components/admin_console/email_settings.jsx]: Message change as requested in PLT-7177

**Comment by jasonblais** [comment]
@MusikPolice Is there something specific you want a PM to review?

**Comment by MusikPolice** [comment]
@jasonblais I was told that all UI changes needed PM review. In this case, there are language changes in the System Console, and the default option in the Account Settings > Notifications screen has changed if batching has been activated.

**Comment by jasonblais** [comment]
For smaller improvements, I think it's fine to queue directly to dev review. If you have a JIRA ticket assigned, at least one PM has reviewed it. So if you implemented according to spec, no PM review is needed.

However, if you have questions, or would like a PM to test/review the PR, that's fine too. Especially on bigger features.

^That's my take on it, but maybe something to ask from devs

**Comment by jasonblais** [comment]
Chatted with @MusikPolice, will do some testing around a potential issue when email batching is disabled on the System Console.

**Comment by jasonblais** [comment]
@MusikPolice I tested the [comments in this JIRA ticket](https://mattermost.atlassian.net/browse/PLT-7177?focusedCommentId=38005&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-38005). I created a separate ticket to look into it, but it doesn't block this PR: https://mattermost.atlassian.net/browse/PLT-7256

I agree it's been an issue for a while, and likely not something that would affect a lot of users.

The main worry I had was that blank account setting fields previously caused the RN app to crash, but I verified this is no longer the case.

File: model/preference.go (hunk 2 above)
Comment (jwilander) [inline model/preference.go]: Maybe don't include "FIFTEEN_MINUTES" in the constant name. If we need to change it for whatever reason (to say 20 minutes), we don't want to have to update the constant name everywhere. I know the client does it but I don't think it's a good idea

**Comment by MusikPolice** [comment]
@jwilander fixed constants naming in https://github.com/mattermost/platform/pull/7036/commits/41d5678f7c316d1e0c2da8bd744ed00749242157

**Pull request merged**